lib/testutils/logrus_hook.go

```diff
@@ -48,8 +47,36 @@ func (smh *SimpleLogrusHook) Lines() []string {
 	return lines
 }
 
+// Reset clears the internal entry buffer.
+func (smh *SimpleLogrusHook) Reset() {
+	smh.mutex.Lock()
+	defer smh.mutex.Unlock()
+	smh.messageCache = []logrus.Entry{}
+}
+
+// LastEntry returns the last entry that was logged (or nil, if no messages were
+// logged or remain).
+func (smh *SimpleLogrusHook) LastEntry() *logrus.Entry {
+	smh.mutex.Lock()
+	defer smh.mutex.Unlock()
+	i := len(smh.messageCache) - 1
+	if i < 0 {
+		return nil
+	}
+	return &smh.messageCache[i]
```

Comment:
This is in `testutils/`, so I am pretty sure we'll have ~0 lock contention most of the time  :sweat_smile: Especially since this only matters for `LastEntry()`, the other methods modify the cache... So soft :-1: for making this more complicated,